spinn_utilities/make_tools/convertor.py

```diff
@@ -0,0 +1,147 @@
+from .file_convertor import FileConvertor
+import os
+import math
+import shutil
+import sys
+
+if 'SPINN_DIRS' in os.environ:
+    RANGE_DIR = os.path.join(os.environ['SPINN_DIRS'], "lib")
+else:
+    RANGE_DIR = "lib"
+
+
+class Convertor(object):
+    # __slots__ = [
+    #    "_dest", "_dest_basename", "_src", "_src_basename"]
+
+    def __init__(self, src, dest, dict):
+        self._src = os.path.abspath(src)
+        if not os.path.exists(self._src):
+            raise Exception(
+                "Unable to locate source directory {}".format(self._src))
+        self._dest = os.path.abspath(dest)
+        src_root, src_basename = os.path.split(
+            os.path.normpath(self._src))
+        dest_root, dest_basename = os.path.split(
+            os.path.normpath(self._dest))
+        if src_root != dest_root:
+            # They must be siblings due to text manipulation in makefiles
+            raise Exception("src and destination must be siblings")
+        else:
+            self._root = src_root
+        self._src_basename = src_basename
+        self._dest_basename = dest_basename
+        self._dict = os.path.abspath(dict)
+
+    def run(self):
+        self._mkdir(self._dest)
+        with open(self._dict, 'w') as dict_f:
+            dict_f.write("Id,Preface,Original\n"
+                         ",DO NOT EDIT,THIS FILE WAS AUTOGENERATED BY MAKE\n")
+        message_id = self._get_id()
+        for dirName, subdirList, fileList in os.walk(self._src):
+            self._mkdir(dirName)
+            for file_name in fileList:
+                _, extension = os.path.splitext(file_name)
+                source = os.path.join(dirName, file_name)
+                if extension in [".c", ".cpp", ".h"]:
+                    destination = self._any_destination(source)
+                    message_id = FileConvertor.convert(
+                        source, destination, self._dict, message_id)
+                elif file_name in ["common.mk", "Makefile.common",
```

Comment:
id create this as a list at top, as then its easier to find for future additions. named skippable_files

Reply:
ok created a CONSTANT